__tests__/components/AnonymousChat/child/ChatItemTargetText.test.tsx

```diff
@@ -1,34 +1,84 @@
+/* eslint-disable global-require */
 import * as React from 'react';
-import {render} from '@testing-library/react-native';
+import FastImage from 'react-native-fast-image';
+import {fireEvent, render} from '@testing-library/react-native';
 
 import ChatItemTargetText from '../../../../src/components/AnonymousChat/child/ChatItemTargetText';
+import {Context} from '../../../../src/context';
+import {DEFAULT_PROFILE_PIC_PATH} from '../../../../src/utils/constants';
+import {SIGNED} from '../../../../src/hooks/core/constant';
 
-describe('TESTING ChatItemTargetText', () => {
-  it('RENDER not continuous should match snapshot', () => {
-    const {toJSON} = render(
+jest.spyOn(React, 'useRef').mockReturnValue({current: {close: jest.fn()}});
+
+const avatar = (
+  <FastImage
+    style={{height: 24, width: 24, borderRadius: 12}}
+    source={{uri: DEFAULT_PROFILE_PIC_PATH}}
+  />
+);
+
+let setReplyPreview;
+let contextValue;
+let timestamp;
+
+beforeEach(() => {
+  setReplyPreview = jest.fn();
+  timestamp = new Date().toISOString();
+  contextValue = {
+    chat: [{replyTarget: null}, setReplyPreview],
+    profile: [{myProfile: {}}]
+  };
+});
+
+afterEach(() => {
+  jest.restoreAllMocks();
+});
+
+const renderChatItemTargetText = () =>
+  render(
+    <Context.Provider value={contextValue}>
       <ChatItemTargetText
-        avatar="https://www.google.com"
+        avatar={avatar}
         username="username"
         isContinuous={false}
         message="message"
-        time="time"
+        time={timestamp}
+        chatType={SIGNED}
+        messageType="regular"
+        data={{
+          id: 'id986',
+          user: {username: 'username'},
+          message: 'message',
+          message_type: 'regular',
+          updated_at: timestamp,
+          chatType: SIGNED
+        }}
       />
-    );
+    </Context.Provider>
+  );
 
-    expect(toJSON()).toMatchSnapshot();
+describe('TESTING ChatItemTargetText', () => {
+  it('should call setReplyPreview when swiped to the left', () => {
+    const {getByTestId} = renderChatItemTargetText();
+    const expected = {
+      payload: {
+        id: 'id986',
+        message: 'message',
+        message_type: 'regular',
+        chatType: 'SIGNED',
+        updated_at: timestamp,
+        user: {username: 'username'}
+      },
+      type: 'SET_REPLY_TARGET'
+    };
+    fireEvent(getByTestId('swipeable'), 'onSwipeableOpen', 'left');
+    expect(setReplyPreview).toHaveBeenCalledWith(expected);
   });
 
-  it('RENDER continuous should match snapshot', () => {
-    const {toJSON} = render(
-      <ChatItemTargetText
-        avatar="https://www.google.com"
-        username="username"
-        isContinuous={true}
-        message="message"
-        time="time"
-      />
-    );
-
-    expect(toJSON()).toMatchSnapshot();
+  it('should match snapshot', () => {
+    const tree = renderChatItemTargetText().toJSON();
+    expect(tree).toMatchSnapshot();
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The changes in this hunk are mostly about adding more tests and improving the existing ones. The new tests cover additional functionalities like swipe actions and reply previews. The use of context and mock functions is appropriate for simulating different scenarios. However, there's a potential issue with the `timestamp` variable. It's defined once in the `beforeEach` block and used across multiple tests. If these tests take longer than a millisecond to run, they could potentially fail due to the timestamp mismatch.

```diff
beforeEach(() => {
  setReplyPreview = jest.fn();
- 28:   timestamp = new Date().toISOString();
+ 28:   timestamp = '2023-12-13T00:00:00.000Z';
  contextValue = {
    chat: [{replyTarget: null}, setReplyPreview],
    profile: [{myProfile: {}}]
  };
});
```

In the above fix, I've replaced the dynamic timestamp with a static one. This ensures that the timestamp value remains consistent across all tests. If your application logic requires testing with dynamic timestamps, consider mocking the Date object to return a fixed timestamp during tests.


<!-- This is an auto-generated comment by OSS CodeRabbit -->